packages/cohort/src/core.rs

```diff
@@ -288,8 +288,16 @@ impl Cohort {
 
         // Talos gave "go ahead"
 
+        // Check safepoint condition before installing ...
+
+        let safepoint = resp.safepoint.unwrap(); // this is safe for 'Committed'
+        SnapshotApi::await_until_safe(Arc::clone(&self.database), safepoint).await?;
+
+        // install
+
         op_impl(Arc::clone(&self.database), AccountRef::new(account.number.clone(), None), amount, resp.version).await?;
 
+        // TODO: this should be done by replicator
         SnapshotApi::update(Arc::clone(&self.database), resp.version).await?;
```

Comment:
Initiator should not update `snapshot` as initiator does out of order install.

Reply:
You already have a comment in your code, so ignore this :-)

Reply:
ok